prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -248,15 +253,37 @@ object CustomFHIRFunctions : FhirPathFunctions {
                 else -> throw SchemaException("Call with day, month, or year")
             }
         } else {
-            calculateAgeWithAssumption(date)
+            calculateAgeWithAssumption(date, comparisonDate)
+        }
+    }
+
+    private fun getParamValue(params: List<Base>): Pair<LocalDate, String?> {
+        val pair: Pair<LocalDate, String?> = Pair(LocalDate.now(), null)
+        val firstParamPair = setTheParamValue(params[0], pair)
+
+        if (params.size == 2) {
+            return setTheParamValue(params[1], firstParamPair)
+        }
+
+        return firstParamPair
+    }
+
+    private fun setTheParamValue(paramValue: Any, pair: Pair<LocalDate, String?>): Pair<LocalDate, String?> {
+        val periodsOfTime = listOf("day", "month", "year")
+        return if (paramValue is DateType) {
+            pair.copy(LocalDate.of(paramValue.year, paramValue.month + 1, paramValue.day), pair.second)
+        } else if (paramValue is StringType && paramValue.toString().lowercase() in periodsOfTime) {
+            pair.copy(pair.first, paramValue.toString())
+        } else {
+            pair
         }
     }
 
-    private fun calculateAgeWithAssumption(date: DateType): MutableList<Base> {
+    private fun calculateAgeWithAssumption(date: DateType, comparisonDate: LocalDate): MutableList<Base> {
         val period = Period.between(
             // have to do plus one for the month because it is expecting 1 based and we get zero based
             LocalDate.of(date.year, date.month + 1, date.day),
```

Comment:
We do this `LocalDate.of(x.year, x.month + 1, x.day)` a lot in this file, it might be helpful to extract that into a one-line helper to make sure the `+ 1` is never missed on the month, and to only have to explain that in one place.